File: scala/hamming/src/main/scala/Hamming.scala
/**
  * Created by chaecramb on 04/08/2016.
  */

object Hamming {
  def compute(strand1: String, strand2: String): Int = {
    require (strand1.length == strand2.length, "Strands must be of the same length") 
    strand1.zip(strand2).count(nucs => nucs._1 != nucs._2)
  }
}

File: ruby/hamming/hamming.rb
module BookKeeping
  VERSION = 3 # Where the version number matches the one in the test.
end

class Hamming

  def self.compute(strand_1, strand_2)
    unless strand_1.size == strand_2.size
      raise ArgumentError.new('strands must be of the same length')
    end
      
    strand_1.split('').zip(strand_2.split('')).reduce(0) do |mutations, nucs|
      mutations + (nucs[0] == nucs[1] ? 0 : 1)
    end
  end

end